sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2612,6 +2614,9 @@ object Sequence {
       val stepDays = step.days
       val stepMicros = step.microseconds
 
+      require(scale != MICROS_PER_DAY || stepMonths != 0 || stepDays != 0,
+        "sequence step must be a day interval if start and end values are dates")
+
       if (stepMonths == 0 && stepMicros == 0 && scale == MICROS_PER_DAY) {
```

Comment:
we can remove this branch now?

Reply:
Seems we need the `require` check in eval, and I remove `SPARK-32198` branch code from this PR, is it ok ?

Reply:
@TJX2014 . What is SPARK-32198? It's not created yet.
> I remove SPARK-32198 branch code from this PR

Reply:
Also, I have the same question like @cloud-fan .

Reply:
@dongjoon-hyun  Sorry, the correct PR is [https://github.com/apache/spark/pull/28856](), this PR is forbid step and the other is cross the timezone.